### PR TITLE
Fix device dispatch for linear updater.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
Still need to update documents

File: src/linear/updater_coordinate.cc [hunk 1]
```diff
@@ -40,6 +40,7 @@ class CoordinateUpdater : public LinearUpdater {
     FromJson(config.at("coordinate_param"), &cparam_);
   }
   void SaveConfig(Json* p_out) const override {
+    LOG(INFO) << "Save config for CPU updater.";
```
Comment (hcho3) [inline src/linear/updater_coordinate.cc:43]: Please remove this line

File: src/linear/updater_gpu_coordinate.cu [hunk 2]
```diff
@@ -42,6 +41,7 @@ class GPUCoordinateUpdater : public LinearUpdater {  // NOLINT
     FromJson(config.at("coordinate_param"), &coord_param_);
   }
   void SaveConfig(Json* p_out) const override {
+    LOG(INFO) << "Save config for GPU updater.";
```
Comment (hcho3) [inline src/linear/updater_gpu_coordinate.cu:44]: Remove this line too

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: src/linear/updater_coordinate.cc (hunk 1 above)
Comment (trivialfis) [reply src/linear/updater_coordinate.cc:43]: I use it for testing, will change it into log debug

Comment (trivialfis) [reply src/linear/updater_coordinate.cc]: done.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**